shared/types/icon.type.ts

```diff
@@ -0,0 +1,4 @@
+export interface IIconType {
```

Comment:
I think 'I' is added for interface, and we add **type** for a type. I don't think this is behaving as both interface and type so choose name accordingly like **IIconProps** if iterrface or **IconProps** if type.